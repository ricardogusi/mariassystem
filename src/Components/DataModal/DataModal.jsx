import React from "react";
import styles from "./datamodal.module.css";
import { useState } from "react";


const Modal = ({ setModal }) => {
  

  const [partidas, setPartidas]=useState([{}])

  function removeModal(event) {
    if (event.target === event.currentTarget) {
      setModal(false);
    }
  }

  return (
    <div className={styles.container} onClick={removeModal}>
      <div className={`${styles.modal} ${styles.anime}`}>
        <div className={styles.header}>          
          <button className={styles.headerButton} onClick={removeModal}>X</button>
          <p>Adicione as informações do seu jogo</p>
        </div>
        <div className={styles.form}>
            <h2>Jogo número:</h2>
            <input type="number" min="0" max="999"/>
            <h2>Pontuação:</h2>
            <input type="number" min="0" max="1000"/>
            <button className={styles.cadastrarButton} >Cadastra Partida</button>

        </div>
        
      </div>
    </div>
  );
};

export default Modal;