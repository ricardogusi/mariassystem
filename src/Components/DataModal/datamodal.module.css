.container {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100vh;
  position: fixed;
  background: rgba(0, 0, 0, 0.4);
  z-index: 500;
  top: 0;
  left: 0;
}

.modal {
  position: fixed;
  display: flex;
  justify-content: center;
  height: 85vh;
  width: 100vh;
  z-index: 1000;
  background: rgb(255, 255, 255);
  border-radius: 8px;
  opacity: 0.3;
  transition: 0.3s;
}

.anime {
  opacity: 0;
  transform: scale(0.7);
  animation: anime 0.3s forwards;
}

@keyframes anime {
  to {
    opacity: 1;
    transform: initial;
  }
}

.header {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 50px;
  border: none;
  box-shadow: 5px 5px 10px -7px rgba(0, 0, 0, 0.75);
  border-radius: 8px 8px 0px 0px;
  background: #297894;
  font-size: 1.5rem;
  font-family: "Ranchers", cursive;
  color: #fb962a;
}

.headerButton {
  background: #fb962a;
  position: absolute;
  top: 10px;
  right: 10px;
  border: none;
  cursor: pointer;
  height: 20px;
  width: 20px;
  font-size: 0.7rem;
  color: white;
}

.form {
  display: flex;
  flex-direction: column;  
  text-align: center;
  position: absolute;
  width: 100%;
  height: 75vh;  
  font-family: "Raleway", sans-serif;
  margin-top: 120px;
  color: #297894;
  
}

.form input {
    width: 90px;
    margin: 0 auto;
}

.cadastrarButton {
  margin: 0 auto;
  font-family: "Raleway", sans-serif;
  font-weight: bold;
  margin-top: 80px;
  height: 50px;
  width: 150px;
  border: none;
  border-radius: 5px;
  background: #fb962a;
  cursor: pointer;
  color: white;
}

.cadastrarButton:hover {
    background-color: #ffcb94;
    color: #297894;
    font-weight: bold;
  }
