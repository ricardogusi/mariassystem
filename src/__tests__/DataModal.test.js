
import React from 'react';
import renderer from 'react-test-renderer'
import DataModal from '../Components/DataModal/DataModal.jsx'



it('Renderiza corretamente', () => {
    const tree = renderer
      .create(<DataModal /> )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  

  