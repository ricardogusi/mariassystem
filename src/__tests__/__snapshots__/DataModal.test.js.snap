// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderiza corretamente 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <div
    className="modal anime"
  >
    <div
      className="header"
    >
      <button
        className="headerButton"
        onClick={[Function]}
      >
        X
      </button>
      <p>
        Adicione as informações do seu jogo
      </p>
    </div>
    <div
      className="form"
    >
      <h2>
        Jogo número:
      </h2>
      <h2>
        # 
      </h2>
      <h2>
        Pontuação:
      </h2>
      <input
        autoFocus={true}
        max="999"
        min="0"
        type="number"
      />
      <button
        className="cadastrarButton"
      >
        Cadastra Partida
      </button>
    </div>
  </div>
</div>
`;
